core/designsystem/src/main/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/component/scrollbar/Scrollbar.kt

```diff
@@ -320,84 +304,123 @@ fun Scrollbar(
                 }
             },
     ) {
-        val scrollbarThumbMovedDp = max(
-            a = with(localDensity) { thumbMovedPx.toDp() },
-            b = 0.dp,
-        )
         // scrollbar thumb container
         Box(
             modifier = Modifier
                 .align(Alignment.TopStart)
-                .run {
-                    when (orientation) {
-                        Orientation.Horizontal -> width(thumbSizeDp)
-                        Orientation.Vertical -> height(thumbSizeDp)
+                .layout { measurable, constraints ->
+                    val state = state.value
+                    val thumbSizePx = max(
+                        a = state.thumbSizePercent * track.size,
+                        b = minThumbSize.toPx(),
+                    )
+
+                    val thumbTravelPercent = when {
+                        interactionThumbTravelPercent.isNaN() -> state.thumbMovedPercent
+                        else -> interactionThumbTravelPercent
+                    }
+
+                    val thumbMovedPx = min(
+                        a = track.size * thumbTravelPercent,
+                        b = track.size - thumbSizePx,
+                    )
+
+                    val scrollbarThumbMovedDp = max(
```

Comment:
The unit here is still pixels instead of dp, so maybe the val name should be `scrollbarThumbMovedPx`?